src/Verification.php

```diff
@@ -39,8 +39,28 @@ public function isValid()
     private function signatureMatches()
     {
         try {
-            $random = random_bytes(32);
-            return hash_hmac('sha256', $this->expectedSignatureBase64(), $random, true) === hash_hmac('sha256', $this->providedSignatureBase64(), $random, true);
+            $key = $this->key();
+            switch ($key->getType()) {
+              case 'secret':
+                $random = random_bytes(32);
+                $expectedResult = hash_hmac(
+                  'sha256', $this->expectedSignatureBase64(), $random, true);
+                $providedResult = hash_hmac(
+                  'sha256', $this->providedSignatureBase64(), $random, true);
```

Comment:
source formatting looks off here too - 4 space indentation, and inline would be easier to read here

Reply:
Fixed in 665baabfcb62596b838b3c852613f45b81be700e